packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts

```diff
@@ -271,3 +281,76 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    */
   componentId?: string;
 }
+
+export type ITextFieldStyleProps = Required<Pick<ITextFieldProps, 'theme'>> &
+  Pick<
+    ITextFieldProps,
+    'className' | 'disabled' | 'required' | 'multiline' | 'borderless' | 'resizable' | 'underlined'
+  > & {
+    /** Element has an error message. */
+    hasErrorMessage?: boolean;
+    /** Element has an icon. */
+    hasIcon?: boolean;
+    /** Element has a label. */
+    hasLabel?: boolean;
+    /** Element has focus. */
+    focused?: boolean;
+  };
+
+export interface ITextFieldSubComponentStyles {
+  // TODO: this should be the interface once we're on TS 2.9.2 but otherwise causes errors in 2.8.4
+  // label: IStyleFunctionOrObject<ILabelStyleProps, ILabelStyles>;
+  label: IStyleFunctionOrObject<any, any>;
```

Comment:
I've confirmed this isn't an issue with TS 2.9.2. I tried to find another way to avoid this, particularly since it's public, but could not without forcing a bunch of cascading anys all throughout TextField. If anybody sees an easier solution until we can use a newer version of TS, let me know.

Reply:
@dzearing @kenotron FYI